### ISSUE TITLE
Problem with tikzupper and tikzlower options

### ISSUE THREAD

**Issue by nignasi** [T-F-S/tcolorbox#40]
tikzlower and tikzupper options fail when more than one parameter is included in their options. 
Something like `tikzupper={red, line width=2pt}` fails because it says `line width` is not recognized while `tikzupper=red` works as expected. As two parameters are not allowed, it's not possible to fix alternative dimensions to `x` and `y`. But this is an special case because when only one is introduced, latex hangs with an error message. Using a `TikZ` style which includes all options works as expected, but I think this is not commented in documentation. Problem reported at https://tex.stackexchange.com/questions/423183/tcolorbox-tikz-picture-option-of-tikz-lower/423198#423198 

```
\documentclass{article}
\usepackage[most]{tcolorbox}
\usepackage{tikz}

\tikzset{
	mystyle/.style={x=3cm, y=2cm, red, line width=2pt}}
	
\begin{document}

%This works. Only one parameter.
\begin{tcolorbox}[tikz upper={red}]
\draw (0,0) rectangle (3,2);
\end{tcolorbox}

%%This doesn't recognize second parameter
%\begin{tcolorbox}[tikz upper={red, line width=1mm}]
%\draw (0,0) rectangle (3,2);
%\end{tcolorbox}

%%This doesn't recognize second parameter
%\begin{tcolorbox}[tikz upper={x=3cm, y=2cm}]
%\draw (0,0) rectangle (3,2);
%\end{tcolorbox}

%%This one hangs (also with only "y")
%\begin{tcolorbox}[tikz upper={x=2cm}]
%\draw (0,0) rectangle (3,2);
%\end{tcolorbox}

%This one works. All parameters in tikz-style.
\begin{tcolorbox}[tikz upper={mystyle}]
\draw (0,0) rectangle (3,2);
\end{tcolorbox}

\end{document}

```

**Comment by T-F-S** [comment]
I seems that the problem can be solved by adding a further pair of brackets to the options for the internal `tikzpicture`, i.e. `\begin{tikzpicture}[{#1}]` instead of `\begin{tikzpicture}[#1]`.

The following patch works for me:

```latex
\makeatletter
\tcbset{
  tikz upper/.style={before upper=\centering\tcb@shield@externalize\begin{tikzpicture}[{#1}],after upper=\end{tikzpicture}},%
  tikz lower/.style={before lower=\centering\tcb@shield@externalize\begin{tikzpicture}[{#1}],after lower=\end{tikzpicture}},%
}
\makeatother
```

I will add this patch to tcolorbox for the next version (which may take a good while).

**Comment by T-F-S** [comment]
Solved with version 4.14 (2018-07-26)

**Comment by T-F-S** [comment]
Sorry, not solved, because of poor bug fixing. This is my next and hopefully final patch (will be integrated into the next version):

```latex
\makeatletter
\tcbset{
  tikz upper/.style={before upper={\centering\tcb@shield@externalize\begin{tikzpicture}[{#1}]},after upper=\end{tikzpicture}},%
  tikz lower/.style={before lower={\centering\tcb@shield@externalize\begin{tikzpicture}[{#1}]},after lower=\end{tikzpicture}},%
}
\makeatother
```

**Comment by T-F-S** [comment]
Solved (finally?) with version 4.15 (2018-12-07)
